### DISCUSSION TITLE
Integrate GitHub CLI into Git Bash (or other terminal) - Use GitHub with your command line

### DISCUSSION THREAD

**Discussion by BrysonMG** [nss-day-cohort-47/C47-Discussions#62] [Look what I learned!]
So I found out that GitHub has a CLI (Command Line Interface) that includes commands that allow you to handle the entire GitHub workflow via your command line. Example: you can create or edit a pull request without ever opening GitHub in your browser, just do it all from the command line.

To get started, go over to https://cli.github.com/ and scroll down a bit to find the download button. I am on Windows, and I was able to just use the install wizard to get it installed. Mac users should be able to do the same.

After it's installed, open your terminal of choice and head over to https://cli.github.com/manual/ to see how to get started (connecting your account to the CLI, etc.). This page also has a list of the commands you can use to control GitHub.

Your first step should be to run the command `gh auth login`. It will give a few prompts inside the terminal to get you set up. Navigate with arrow keys and enter button.

1. "What account do you want to log into?" **GitHub.com**
2. "What is your preferred protocol for Git operations?" **SSH**
3. "Upload your SSH public key to your GitHub account?" **Skip** (you can skip this since we already added our public key during the prework/first week of class). If you need to add a key to your account, you can do it here.
4. "How would you like to authenticate GitHub CLI?" **Login with a web browser**
5. Copy the one-time code given in the terminal, and press Enter.
6. A page will open in your browser to input the one-time code, paste it in and authorize GitHub CLI.
7. Go back to your terminal and hit Enter one more time. and it should tell you that you're logged in.

**Comment by coltmay** [comment]
Congratulations, you've potentially just started your journey down the path of [Vim](https://www.vim.org/).

Embrace the CLI, embrace it.

**Reply by BrysonMG** [reply]
Maybe lol. I typically approach this stuff slowly and cautiously. Curious to see where I end up months from now.

**Reply by coltmay** [reply]
If you stay on the pace you've been on, I'd say you're doing awesome.
